composer.json

```diff
@@ -5,7 +5,8 @@
     "require": {
         "psr/log": "^1.0.2",
         "clue/graph": "0.9.0",
-        "graphp/algorithms": "0.8.1"
+        "graphp/algorithms": "0.8.1",
+        "bitexpert/slf4psrlog": "^0.1.3"
```

Comment:
Where did you find this library? I haven't heard of it. Why aren't we just depending on `psr/log`?

Reply:
I found it in the comments on your logger injection blog post. I tried to summarize the reasons for using it in the description of this pull request. I'm not married to this approach, so let me know if you have a better way in mind.

Reply:
I re-read your comment.  I think all the things you think are wins for this logger factory are actually solved by the "Optional setLogger Method" from my article.  By default we use NullLogger which only depends on the PSR-3 interface package (not a concrete implementation). If they want to pass in a real logger that's fine.